### ISSUE TITLE
how to pre-train on a custom dataset ? 

### ISSUE THREAD

**Issue by StephennFernandes** [microsoft/SpeechT5#1]
Hey there, I am looking forward. to pre-training SpeechT5 on a custom dataset. preferably multi-lingual datasets. could i please get some references, documentations etc as a starting point to get started on the same please. Thanks.

**Comment by Ajyy** [comment]
Hi,

Please follow the steps in [here](https://github.com/microsoft/SpeechT5/tree/main/SpeechT5#data-preparation) to prepare the data.

You may also need to extract HuBERT label on your custom dataset and prepare the speaker embedding (by using the multi-lingual speaker model). After that, you can follow our [scripts ](https://github.com/microsoft/SpeechT5/blob/main/SpeechT5/README.md#pre-training) to train the model.

**Comment by StephennFernandes** [comment]
@Ajyy , I see on your Huggingface account you have uploaded speechT5 to Huggingface is it functional yet ?

**Comment by StephennFernandes** [comment]
@Ajyy by any chance would you be releasing speechT5 to Huggingface ?

**Comment by StephennFernandes** [comment]
@Ajyy 

Hi, 
Can you show me any resources and links where i can extract HuBERT label on my custom dataset and prepare the speaker embeddings, 
FYI i have a multi-lingual dataset

**Comment by StephennFernandes** [comment]
Hey @Ajyy Thanks a ton for replying back ! means a lot

**Comment by StephennFernandes** [comment]
@Ajyy 

I have noitced that for hubert extraction there is not multilingual HuBERT model available. so should i use the existing english HuBERT model ? would this affect my multilingual speechT5 models performance during pretraining and finetuning ?

**Comment by Ajyy** [comment]
Hi, I think you can try to use [mHuBERT](https://github.com/facebookresearch/fairseq/blob/c8a8e2c39204cffff082be508bbfd60254f5f1b7/examples/speech_to_speech/docs/textless_s2st_real_data.md#hubert) for multilingual SpeechT5. The english HuBERT model may affect the performance.

**Comment by StephennFernandes** [comment]
@Ajyy thanks a ton, using mHuBERT now for extracting HuBERT feature

**Comment by StephennFernandes** [comment]
@Ajyy how could i obtain xvectors on my training data ? I tried #16 but thats not working

**Comment by StephennFernandes** [comment]
@Ajyy okay got it!

 thanks a ton

**Comment by StephennFernandes** [comment]
@Ajyy when extracting hubert labels whats the ideal `n_cluster` value to be set ?

**Comment by StephennFernandes** [comment]
@Ajyy also slightly confused about the fairseq-preprocess script, could you show me how do i work around that, i have a larrge `train.txt` and `valid.txt` file for which i have used `spm.model` to encode them into `train_encoded.txt` and `valid_encoded.txt`  

But i am confused about what it means about a dictionary creation and .bin files and how to feed then into `fairseq-preprocess`

**Comment by Ajyy** [comment]
> @Ajyy also slightly confused about the fairseq-preprocess script, could you show me how do i work around that, i have a large `train.txt` and `valid.txt` file for which i have used `spm.model` to encode them into `train_encoded.txt` and `valid_encoded.txt`
> 
> But i am confused about what it means about a dictionary creation and .bin files and how to feed then into `fairseq-preprocess`

You can check the [preprocess](https://github.com/facebookresearch/fairseq/tree/main/examples/language_model#1-preprocess-the-data) for language model of fairseq.

After preprocess by fairseq-preprocess, you will get a bin and index file for your dataset, which can be read faster compared to txt file.
